esp-hal/src/dma/mod.rs

```diff
@@ -2405,6 +2508,56 @@ impl DmaTxRxBuf {
     }
 }
 
+impl DmaTxBuffer for DmaTxRxBuf {
```

Comment:
Slightly orthogonal to this PR, but is there any reason why `DmaTxRxBuf` isn't defined as

```rust
struct DmaTxRxBuf {
	rx: DmaRxBuf,
    tx: DmaTxBuf,
}
```

This would reduce the duplicate checks on construction, and avoid duplicate(ish) trait impls here.

Reply:
also as per the new API guidelines, can you rename to `DmaRxTxBuf`?

Reply:
To share the buffer basically, `DmaTxRxBuf` is one buffer accessed via two lists of descriptors.
Using both `DmaRxBuf` and `DmaTxBuf` would mean having two buffers.

> also as per the new API guidelines, can you rename to `DmaRxTxBuf`?

Bahhh, I forgot about the name (I guess that should've been done in the RX-TX PR). Sure I'll rename it, np.

Reply:
> To share the buffer basically

Of course, sorry for the noise!